### DISCUSSION TITLE
System and Library not showing in web interface

### DISCUSSION THREAD

**Discussion by stymaster** [rern/rAudio#1748] [Fixed Issues]
Hi. I'm using rAudio-RPi2-20240818.img.xz on an rPi2, and the System page never loads- just the flashing logo in the centre. The Library shortcut at the bottom only worked once, too- the page never loads. The library is visible from an MPD client though. 
This first happened after an upgrade from within the UI, but I've re-imaged as well. My library isn't massive and is on an external USB hdd. Any ideas? I have SSH access.

**Comment by Sagnasty** [comment]
I am also having a similar issue on four separate pi's after upgrading to rAudio 20240818.   In my particular case, the library page will not show up (I can see the system page however).   I confirmed that the NAS library NFS mount was active, MPD was running and showed the correct library information.  The main and  playlist are displayed correctly.   When the library page is selected ("+" or bottom library tab), the last active page will be continue to be displayed and the bottom active tab will switch to the library tab.

I also tried applying rAudio-64bit-20240818.img and rebuilding one system, but the exact same behavior occurred.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Reply by Sagnasty** [reply]
Behavior persists in rAudio 20240906

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by timw625** [comment]
Have tried a clean re-image from rAudio-64bit-20240818.img.xz and am also not able to see the system page. Once again it just leave the flashing wait image in the middle od screen.
Checksum of downloaded image file matched that published.
Updated to 20240908 via addons but problem still there.
Same results on USB or flash drives

Library seems to work for me

Rapberry PI 4b + SMSL dac

**Comment by hjfisch** [comment]
For me both are working .
Connected 4tb USB-SSD , Raspberry 3B+ , Allo Digione

**Comment by Sagnasty** [comment]
If you are having an issue with the library page not being displayed, #1765 may apply to you.
I have not had an issue with the systems page showing up.

**Reply by timw625** [reply]
After a bit more playing I found a flash card with raudio-1 version that did not ask for a re-image to update.
HAving taken a backup and a dit copy of the card I patched that to the latest verson and it works fine
HOWEVER
Building from the image file as I mentioned above does not work.

I have notices that the player versons are different
on the upgraded working version - 0.23.15-2
on the re-imaged build - 0.23.15-5
Both running 20240908. Only differeance one upgraded other built from image

I tried copying the http directory structure accross from working to broken as I saw some differances in file but this dd not help.

Library pages working on both builds for me. Just system will not display

**Reply by timw625** [reply]
I normally use Firefox browser and if I try to access the System Tab on the reimaged build I still get the flashing raudio icon and nothing else.

If I try to access with Chrome I get a couple of errors:

     Errors: Unexpected token 'o', ..."        : off
     , "ico"... is not valid JSON

and a page of text that I do not understand.
Not surprisingly Edge also gives the same error and page of test.

On the working system Chrome and edge displays the system page without issue. 

Any thoughts as to where to look next much appriciated

**Reply by rern** [reply]
Try without restoring the backup. Too old of a backup contains some incompatible data and settings.

A backup should be used when it's made from an up to date rAudio.

**Reply by timw625** [reply]
many thanks rern
On the re-imaged build I am having the issue with at present, I have not restored a backup file to it.
It is literally just the image copied to the flash card or usb using the latest RPI imager and a previous version for good measure.
Connected via WiFi initially then configured static LAN address.
Same results across either medium.

Have tried different flash and USB but again same result with re-image.

Only a clone of previous build I updated works.

It don't make sense has to be something stupid but I can't figure out what.

**Reply by rern** [reply]
Please SSH to rAudio and get the output of: `/srv/http/bash/settings/system-data.sh`

**Comment by timw625** [comment]
[system-dat.sh output.txt](https://github.com/user-attachments/files/16993492/system-dat.sh.output.txt)

Again Many thanks for your assistance.
Hopefully I have attached a txt file with a grab of the output.
If not please let me know and I will paste into this window

All the best

**Reply by rern** [reply]
Please try:
`sed -i 's/apm=false/& || apm=true/' /srv/http/bash/settings/system-storage.sh`

**Comment by timw625** [comment]
Thankyou so much!!!!

The System page now comes up.
I will now complete my build.

Can you give me a simple explanation of what your command did. 
Did it change the status of apm from false to true and then re-run systen-storage.sh?
What is apm?

Again thanks. All the best

**Reply by rern** [reply]
It's the current status of `apm` Advanced Power Management for USB drives which is `off`, not valid as boolean. Prior to this, it's blank value which was converted to `false`.
